bundles/org.openhab.binding.dirigera/src/main/java/org/openhab/binding/dirigera/internal/handler/light/TemperatureLightHandler.java

```diff
@@ -0,0 +1,138 @@
+            while (attributesIterator.hasNext()) {
+                String key = attributesIterator.next();
+                if ("colorTemperatureMin".equals(key)) {
+                    colorTemperatureMin = attributes.getInt(key);
+                    properties.put("colorTemperatureMin", String.valueOf(colorTemperatureMin));
+                } else if ("colorTemperatureMax".equals(key)) {
+                    colorTemperatureMax = attributes.getInt(key);
+                    properties.put("colorTemperatureMax", String.valueOf(colorTemperatureMax));
+                }
+            }
+            updateProperties(properties);
+            range = colorTemperatureMin - colorTemperatureMax;
+            handleUpdate(values);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        super.handleCommand(channelUID, command);
+        String channel = channelUID.getIdWithoutGroup();
+        String targetProperty = channel2PropertyMap.get(channel);
+        if (targetProperty != null) {
+            switch (channel) {
+                case CHANNEL_LIGHT_TEMPERATURE:
+                    if (command instanceof PercentType percent) {
+                        /*
+                         * some color lights which inherit this temperature light don't have the temperature capability.
+                         * As workaround child class ColorLightHandler is handling color temperature
+                         */
+                        if (receiveCapabilities.contains(Model.COLOR_TEMPERATURE_CAPABILITY)) {
+                            int kelvin = getKelvin(percent.intValue());
+                            JSONObject attributes = new JSONObject();
+                            attributes.put(targetProperty, kelvin);
+                            super.changeProperty(LightCommand.Action.TEMPERARTURE, attributes);
+                            if (!isPowered()) {
+                                // fake event for power OFF
+                                updateState(channelUID, percent);
+                            }
+                        }
+                    } else if (command instanceof OnOffType onOff) {
+                        super.addOnOffCommand(OnOffType.ON.equals(onOff));
+                    }
+                    break;
+            }
+        }
+    }
+
+    @Override
+    public void handleUpdate(JSONObject update) {
+        super.handleUpdate(update);
+        if (update.has(Model.ATTRIBUTES)) {
+            JSONObject attributes = update.getJSONObject(Model.ATTRIBUTES);
+            Iterator<String> attributesIterator = attributes.keys();
+            while (attributesIterator.hasNext()) {
+                String key = attributesIterator.next();
+                String targetChannel = property2ChannelMap.get(key);
+                if (targetChannel != null) {
+                    switch (targetChannel) {
+                        case CHANNEL_LIGHT_TEMPERATURE:
```

Comment:
Many other light bindings provide TWO color-temperature channels. i.e. Color_Temperature_Percent and Color_Temperature_Absolute. Again see my example in Hue API v2. Having the latter Color_Temperature_Absolute channel allows OH Main UI / Sitemap UI to utilise the ColorTemperaturePicket widget that visually displays the actual range of min/max color temperature (provided that the channel has the correct state description min/max values).

Reply:
Done